dts/bindings/clock/nordic,nrf-global-hsfll.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic Global HSFLL clock.
+
+  The lowest supported clock frequency is the default
+  clock frequency.
+
+  Example:
+
+    global_hsfll: global_hsfll {
+            compatible = "nordic,nrf-global-hsfll";
+            clocks = <&fll16>;
+            #clock-cells = <0>;
+            supported-clock-frequencies = <64000000
+                                           128000000
+                                           256000000
+                                           320000000>;
+    };
+
+compatible: "nordic,nrf-global-hsfll"
```

Comment:
I suggest `nordic,nrf-hsfll-global` to align with other Nordic bindings. Maybe even rename the local domain HSFLL binding to `nordic,nrf-hsfll-local`.

Reply:
I would like to do that, what about renaming `hsfll120:` to `hsfll_global:` in the devicetree?

Reply:
It can stay as `hsfll120` to match the datasheet.

Reply:
deal, and done

Reply:
Okay, you used `nordic,nrf-local-hsfll` instead?

Reply:
oh, sorry, did not notice the change of order :) will update

Reply:
note: there are multiple global HSFLLs: HSFLL120, 121, 200

Reply:
so far, only HSFLL120 is supported by GDFS, so any other clock is effectively a "fixed-clock" for now

Reply:
Sure, my comment was intended to guide on correct nodes, bindings, and compatibles naming

Reply:
Are there nodes which are misnamed currently that need to be addressed in this  PR?

Reply:
I don't know. I didn't review this PR. I only noticed you discussed renaming `hsfll120` to `hsfll_global` which would be correct only if HSFLL120 was the only global HSFLL instance. I shared a note that there are more instances to highlight that such renaming would be most probably incorrect.

Reply:
ah, makes sense :)

Reply:
> I would like to do that, what about renaming hsfll120: to hsfll_global: in the devicetree?

Sorry about being late, but I agree with @hubertmis here, calling hsfll120 the global hsfll is simplifying things too much. globalhsfll is used throughout the code here, in kconfig etc but only actually means hsfll120 and i think a mass rename from globalhsfll to hsfll120 would be way better